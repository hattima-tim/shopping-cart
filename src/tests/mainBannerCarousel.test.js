import { render, screen } from "@testing-library/react";
import MainBannerCarousel from "../components/mainBannerCarousel";
import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";

describe("first image in carousel", () => {
  test("image should be visible", () => {
    render(<MainBannerCarousel />);
    const image = screen.getByAltText("shop_image");
    expect(image).toBeVisible();
  });

  test("image have text,dots,buttons", () => {
    const { container } = render(<MainBannerCarousel />);
    expect(container).toMatchSnapshot();
  });

  test("clicking second dot shows second image", () => {
    render(<MainBannerCarousel />);
    const secondDot = screen.getAllByRole("listitem")[1];
    userEvent.click(secondDot);
    const secondImage = screen.getByAltText("white_background");
    expect(secondImage).toBeVisible();
  });

  test("next button shows second image", () => {
    render(<MainBannerCarousel />);
    const nextButton = screen.getByRole("button", { name: ">" });
    userEvent.click(nextButton);
    const secondImage = screen.getByAltText("white_background");
    expect(secondImage).toBeVisible();
  });
});

describe('auto slide', () => { 
  test('after 5 seconds, the second image should be visible', async() => {
    jest.useFakeTimers();
    render(<MainBannerCarousel />);
    
    act(() => {
      jest.advanceTimersByTime(5000);
    });

    const secondImage = await screen.findByAltText('white_background');
    expect(secondImage).toBeVisible();
  });
  test('after 10 seconds, the first image should be visible', async () => {
    jest.useFakeTimers();
    render(<MainBannerCarousel />);
    
    act(() => {
      jest.advanceTimersByTime(10000);
    });
    
    const firstImage = await screen.findByAltText('shop_image');
    expect(firstImage).toBeVisible();
  });
})

describe("second image in carousel", () => {

  test("clicking first dot shows first image", () => {
    render(<MainBannerCarousel />);
    const secondDot = screen.getAllByRole("listitem")[1];
    userEvent.click(secondDot);
    const firstDot = screen.getAllByRole("listitem")[0];
    userEvent.click(firstDot);
    const firstImage = screen.getByAltText("shop_image");
    expect(firstImage).toBeVisible();
  });

  test("prev button shows first image", () => {
    render(<MainBannerCarousel />);
    const secondDot = screen.getAllByRole("listitem")[1];
    userEvent.click(secondDot);
    const prevButton = screen.getByRole("button", { name: "<" });
    userEvent.click(prevButton);
    const firstImage = screen.getByAltText("shop_image");
    expect(firstImage).toBeVisible();
  });
});