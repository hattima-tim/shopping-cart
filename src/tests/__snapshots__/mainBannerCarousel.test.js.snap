// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first image in carousel image have text,dots,buttons 1`] = `
<div>
  <div
    id="carousel_container"
  >
    <div
      class="carousel_item"
      id="carousel_item_1"
    >
      <img
        alt="shop_image"
        src="https://res.cloudinary.com/du3oueesv/image/upload/v1659705449/shopping%20cart/Rectangle_1_yxbsci.png"
      />
      <ul
        class="nav_dots"
      >
        <li
          class="dot"
          style="background-color: rgb(255, 255, 255);"
        />
        <li
          class="dot"
        />
      </ul>
    </div>
    <div
      class="overlay"
    />
    <div
      class="text"
    >
      <h4>
        SUMMER ARRIVAL
      </h4>
      <h4>
        NEW CUT & SEW T-SHIRTS
      </h4>
    </div>
    <button
      class="view_collection_btn"
    >
      VIEW COLLECTION 
      &gt; &gt;
    </button>
    <div
      id="nav_arrow"
    >
      <button
        id="prev_arrow"
      >
        &lt;
      </button>
      <button
        id="next_arrow"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;
